src/cljs/rems/table.cljs

```diff
@@ -171,48 +172,39 @@
                  ;; performance optimization: hide DOM nodes instead of destroying them
                  (assoc row ::display-row? (display-row? row columns filters))))))))
 
-(defn- focus-async!
-  ([parent selector]
-   (focus-async! parent selector 500 10))
-  ([parent selector timeout interval]
-   (assert (pos? interval))
-   (let [target (.querySelector parent selector)]
-     (cond
-       target (.focus target)
-       (pos? timeout) (js/setTimeout #(focus-async! parent selector (- timeout interval) interval)
-                                     interval)
-       :else (js/console.warn (str "Could not focus element " selector " under:") parent)))))
-
 (defn search [table]
-  (let [filtering @(rf/subscribe [::filtering table])
+  (let [search-params (atom nil)
```

Comment:
It's unfortunate that there needs to be local state here with so specific requirements. Some possible alternatives:
- atom with map from `:id` to element maintained in the focus `ns`
- re-frame atom to contain the element in a path that contains `::focus` and `:id`

Reply:
Using a reagent atom will produce infinite render loop, because setting the ref will re-render the component. It needs to be a `clojure.core/atom` as said in https://cljdoc.org/d/reagent/reagent/0.8.1/doc/frequently-asked-questions/how-do-i-use-react-s-refs-

Reply:
Since one page can contain multiple search components, the key for such a global mapping would need to be e.g. `[(:id table) :search-params]` which I think is more complex than a couple local atoms. It would also be different from the recommended way of using refs in Reagent, so extra justification is needed.